app/src/behaviors/behavior_tri_state.c

```diff
@@ -0,0 +1,301 @@
+            continue;
+        }
+        if (!is_layer_ignored(tri_state, ev->layer)) {
+            LOG_DBG("Tri-State layer changed, ending at %d %d", tri_state->position, ev->layer);
+            tri_state->is_active = false;
+            struct zmk_behavior_binding_event event = {.position = tri_state->position,
+                                                       .timestamp = k_uptime_get()};
+            if (tri_state->is_pressed) {
+                behavior_keymap_binding_released(
+                    (struct zmk_behavior_binding *)&tri_state->config->continue_behavior, event);
+            }
+            behavior_keymap_binding_pressed(
+                (struct zmk_behavior_binding *)&tri_state->config->end_behavior, event);
+            behavior_keymap_binding_released(
+                (struct zmk_behavior_binding *)&tri_state->config->end_behavior, event);
+            return ZMK_EV_EVENT_BUBBLE;
+        }
+    }
+    return ZMK_EV_EVENT_BUBBLE;
+}
+
+#define _TRANSFORM_ENTRY(idx, node)                                                                \
+    {                                                                                              \
+        .behavior_dev = DT_LABEL(DT_INST_PHANDLE_BY_IDX(node, bindings, idx)),                     \
+        .param1 = COND_CODE_0(DT_INST_PHA_HAS_CELL_AT_IDX(node, bindings, idx, param1), (0),       \
+                              (DT_INST_PHA_BY_IDX(node, bindings, idx, param1))),                  \
+        .param2 = COND_CODE_0(DT_INST_PHA_HAS_CELL_AT_IDX(node, bindings, idx, param2), (0),       \
+                              (DT_INST_PHA_BY_IDX(node, bindings, idx, param2))),                  \
+    }
+
+#define IF_BIT(n, prop, i) BIT(DT_PROP_BY_IDX(n, prop, i)) |
+
+#define TRI_STATE_INST(n)                                                                          \
+    static struct behavior_tri_state_config behavior_tri_state_config_##n = {                      \
+        .ignored_key_positions = DT_INST_PROP(n, ignored_key_positions),                           \
+        .ignored_key_positions_len = DT_INST_PROP_LEN(n, ignored_key_positions),                   \
+        .ignored_layers = DT_INST_FOREACH_PROP_ELEM(n, ignored_layers, IF_BIT) 0,                  \
+        .ignored_layers_len = DT_INST_PROP_LEN(n, ignored_layers),                                 \
+        .timeout_ms = DT_INST_PROP(n, timeout_ms),                                                 \
+        .tap_ms = DT_INST_PROP(n, tap_ms),                                                         \
+        .start_behavior = _TRANSFORM_ENTRY(0, n),                                                  \
+        .continue_behavior = _TRANSFORM_ENTRY(1, n),                                               \
+        .end_behavior = _TRANSFORM_ENTRY(2, n)};                                                   \
+    DEVICE_DT_INST_DEFINE(n, behavior_tri_state_init, NULL, NULL, &behavior_tri_state_config_##n,  \
```

Comment:
in #2028 `DEVICE_DT_INST_DEFINE` was migrated to `BEHAVIOR_DT_INST_DEFINE` in behaviours

```suggestion
    BEHAVIOR_DT_INST_DEFINE(n, behavior_tri_state_init, NULL, NULL, &behavior_tri_state_config_##n,  \
```